pkg/probe/probe.go

```diff
@@ -0,0 +1,9 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package probe
+
+import "go.opentelemetry.io/auto/internal/pkg/instrumentation/probe"
+
+// Event represents a probed span event.
+type Event = probe.Event
```

Comment:
I'm hesitant to export this type.

I just noticed that the start/end time are not actually unix timestamps, but the relative offset from startup.

I think we need to better evaluate the structure of this event type before we publish something others will rely on.  It might make the most sense if we actually use a proto representation of spans similar to the collector's `pdata` approach. That way we could avoid deserialization if an exporter speaks that protocol.

Reply:
@MrAlias I pushed an update adding a wrapper struct that gives us a public abstraction from the internal types. Ideally I think we should eventually just take a good look at the internal types and make them public, but something like this could give us a nice bridge to that. Wdyt? Open to ideas